client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,45 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to 'CONSUMPTION' based,
+     * the Reader Group needs to register with the Stream as a Subscriber
+     * and periodically publish retention StreamCuts either manually or at checkpoints.
+     * This can be done by setting retentionConfig to 'CONSUMPTION_BASED_USER_STREAMCUT' or
+     * 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
+     * If the Stream's RetentionPolicy is not Consumption based, retentionConfig should be 'NO_IMPACT'.
+     *
+     * If the Streams' Retention Policy is set to 'CONSUMPTION', but the ReaderGroup retentionConfig is set at
+     * 'NO_IMPACT', the ReaderGroup will not get registered as a Subscriber with the Stream
+     * and retention StreamCuts cannot be published for this ReaderGroup.
```

Comment:
Please do not use the word "published" it has other connotations. 

Additionally there is no need to explain the implementation. Stick to explaining the effect on the user's application.

Reply:
removed the word publish...